service/src/test/java/bio/terra/cbas/controllers/TestRunSetsApiController.java

```diff
@@ -208,7 +210,7 @@ class TestRunSetsApiController {
   @MockBean private BearerToken bearerToken;
   @MockBean private SamClient samClient;
   @MockBean private UsersApi usersApi;
-  @MockBean private SamService samService;
+  @SpyBean private SamService samService;
```

Comment:
just curious - why this change? it's not obvious to me what this is enabling

Reply:
Yeah can you explain why we need `SpyBean` here? 🤔

Reply:
MockBean will stub method implementations (and return some dummy info like `null` or `false`) by default, whereas SpyBean will call the real implementations by default. When mocking something like `samService` that has private/package-private helper methods (`hasPermission`), using `spy` over `mock` ensures that those helper methods can actually be called when needed, like on line 347 of this file.